staker/txbuilder/builder.go

```diff
@@ -30,34 +28,76 @@ type ValidatorWalletInterface interface {
 // This inherits from an ethclient.Client so it can be used to transparently
 // intercept calls to SendTransaction and queue them for the next batch.
 type Builder struct {
-	*ethclient.Client
 	transactions []*types.Transaction
-	builderAuth  *bind.TransactOpts
+	singleTxAuth bind.TransactOpts
+	multiTxAuth  bind.TransactOpts
 	isAuthFake   bool
+	authMutex    sync.Mutex
 	wallet       ValidatorWalletInterface
+	gasRefunder  common.Address
 }
 
-func NewBuilder(wallet ValidatorWalletInterface) (*Builder, error) {
-	randKey, err := crypto.GenerateKey()
-	if err != nil {
-		return nil, err
-	}
-	builderAuth := wallet.AuthIfEoa()
+func NewBuilder(wallet ValidatorWalletInterface, gasRefunder common.Address) (*Builder, error) {
+	var builderAuth bind.TransactOpts
 	var isAuthFake bool
-	if builderAuth == nil {
-		// Make a fake auth so we have txs to give to the smart contract wallet
-		builderAuth, err = bind.NewKeyedTransactorWithChainID(randKey, big.NewInt(9999999))
+	if authIfEoa := wallet.AuthIfEoa(); authIfEoa != nil {
```

Comment:
Just cause I'm trying to learn the code. What's an EOA? Maybe we could put a doc comment on the EOA struct?

Reply:
An EOA is an Externally Owned Account. It's an Ethereum term that means the address has a private key, as opposed to smart contracts.

Reply:
I'd add a comment to this but I'm not sure there's a great place for the comment. There's multiple implementations and multiple interfaces that define it.

Reply:
I'd put it here: https://github.com/OffchainLabs/nitro/blob/master/staker/validatorwallet/eoa.go#L21

That's where I traced the implementation of AuthIfEoa to and found myself going, "Yeah, but what is an EOA?"